docs/migration-guide.md

```diff
@@ -0,0 +1,708 @@
+# Migrating from Unversioned to Versioned Workflows with Temporal Worker Controller
+
+This guide helps teams migrate from unversioned Temporal workflows to versioned workflows using the Temporal Worker Controller. It assumes you are currently running workers without Temporal's Worker Versioning feature and want to adopt versioned worker deployments for safer, more controlled rollouts.
+
+## Important Note
+
+This guide uses specific terminology that is defined in the [Concepts](concepts.md) document. Please review the concepts document first to understand key terms like **Temporal Worker Deployment**, **`TemporalWorkerDeployment` CRD**, and **Kubernetes `Deployment`**, as well as the relationship between them.
+
+## Table of Contents
+
+1. [Why Migrate to Versioned Workflows](#why-migrate-to-versioned-workflows)
+2. [Prerequisites](#prerequisites)
+3. [Understanding the Differences](#understanding-the-differences)
+4. [Migration Strategy](#migration-strategy)
+5. [Step-by-Step Migration](#step-by-step-migration)
+6. [Configuration Reference](#configuration-reference)
+7. [Testing and Validation](#testing-and-validation)
+8. [Common Migration Patterns](#common-migration-patterns)
+9. [Troubleshooting](#troubleshooting)
+
+## Why Migrate to Versioned Workflows
+
+If you're currently running unversioned Temporal workflows, you may be experiencing challenges with deployments:
+
+- **Deployment Risk**: Code changes can break running workflows if they're not backward compatible
+- **Rollback Complexity**: Rolling back deployments can disrupt in-flight workflows
+- **Workflow Determinism Issues**: Changes to workflow logic can cause non-deterministic errors
+
+Versioned workflows with the Temporal Worker Controller solve these problems by:
+
+- ✅ **Safe Deployments**: New versions run alongside old ones, ensuring running workflows complete successfully
+- ✅ **Automated Rollouts**: Progressive rollout strategies reduce risk of new deployments
+- ✅ **Easy Rollbacks**: Can instantly route new workflows back to previous versions
+- ✅ **Workflow Continuity**: Running workflows can continue on their original version until completion
```

Comment:
would it make sense to link out to https://docs.temporal.io/production-deployment/worker-deployments/worker-versioning for the benefits (and if the docs are missing some of the things mentioned here, move that to the docs) instead of rehashing it here?

Reply:
The first section of that doc gives a good overview of the benefits, and I feel like we don't need to repeat ourselves